clar2wasm/src/words/equal.rs

```diff
@@ -88,6 +88,178 @@ impl Word for IsEq {
+            .local_set(end_offset);
+        // STACK: []
+
+        // Traverse the item, leaving it on top of the stack.
+        let item = args.get_expr(1)?;
+        generator.traverse_expr(builder, item)?;
+        // STACK: [item]
+
+        // Get the type of the item expression
+        let item_ty = generator
+            .get_expr_type(item)
+            .expect("index_of item expression must be typed")
+            .clone();
+
+        // Store the item into a local.
+        let item_locals = generator.save_to_locals(builder, &item_ty, true);
+        // STACK: []
+
+        // Create and store an index into a local.
+        let index = generator.module.locals.add(ValType::I64);
+        builder.i64_const(0);
+        // STACK: [0]
+        builder.local_set(index);
+        // STACK: []
+
+        // Loop through the sequence.
+        let loop_body_ty = InstrSeqType::new(
+            &mut generator.module.types,
+            &[],
+            &[ValType::I32, ValType::I64, ValType::I64],
+        );
+        builder.loop_(loop_body_ty, |loop_| {
+            // Loop label.
+            let loop_id = loop_.id();
+
+            // Load an element from the sequence, at offset position,
+            // and push it onto the top of the stack.
+            let elem_size = generator.read_from_memory(loop_, offset, 0, elem_ty);
+            // STACK: [element]
+
+            // Store the current sequence element into a local.
+            let elem_locals = generator.save_to_locals(loop_, elem_ty, true);
+            // STACK: []
+
+            // Check item and element equality.
+            // And push the result of the comparison onto the top of the stack.
+            let _res = wasm_equal(
+                &item_ty,
+                elem_ty,
+                generator,
+                loop_,
+                &item_locals,
+                &elem_locals,
+            );
+            // STACK: [wasm_equal_result]
+
+            loop_.if_else(
+                InstrSeqType::new(
+                    &mut generator.module.types,
+                    &[],
+                    &[ValType::I32, ValType::I64, ValType::I64],
+                ),
+                |then| {
+                    then.i32_const(1).local_get(index).i64_const(0);
+                    // STACK: [1, index_lo, index_hi]
+                },
+                |else_| {
+                    // Increment the sequence offset by the size of the element
+                    // and push it to the stack.
+                    // Also push the offset limit onto the top of the stack.
+                    else_
+                        .local_get(offset)
+                        .i32_const(elem_size)
+                        .binop(BinaryOp::I32Add)
+                        .local_tee(offset)
+                        .local_get(end_offset);
+                    // STACK: [offset, end_offset]
+
+                    else_.binop(BinaryOp::I32GtU).if_else(
```

Comment:
This should be `I32GeU` because once the offset is equal to the end, we don't want to read any more.

Reply:
Done.